### ISSUE TITLE
RedundantModifierCheck only reports first instance of redundant modifier

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#10806]
From https://checkstyle.sourceforge.io/config_modifier.html#RedundantModifier :

> Enums by definition are static implicit subclasses of java.lang.Enum<E>. So, the static modifier on the enums is redundant. In addition, if enum is inside of interface, public modifier is also redundant.

Noticed in https://github.com/checkstyle/checkstyle/pull/10804

```bash
➜  company cat config.xml                                                                           
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>

    <module name="TreeWalker">
        <module name="RedundantModifier">
<!--         <property name="tokens" value="STRICTFP"/>-->
    </module>

    </module>
</module>

➜  company cat Test.java 
interface Test {
    public enum MyInnerEnum1 {} // violation
    static enum MyInnerEnum2 {} // violation
    public static enum MyInnerEnum3 {} // should be 2 violations
    static public enum MyInnerEnum4 {} // should be 2 violations
}

➜  company javac Test.java 

➜  company java -jar checkstyle-9.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:2:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:3:5: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:4:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:5:5: Redundant 'static' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 4 errors.

```
I would expect checkstyle to report all redundant modifiers, like this:

```bash

➜  company java -jar checkstyle-9.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:2:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:3:5: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:4:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:4:12: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:5:5: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /home/nick/IdeaProjects/Tester/src/com/company/Test.java:5:12: Redundant 'public' modifier. [RedundantModifier
Audit done.
Checkstyle ends with 4 errors.

```

**Comment by AkMo3** [comment]
I am on it. It is actually due to a break statement.
https://github.com/checkstyle/checkstyle/blob/acf991a1ca8a44d6975ab81ed59e98b558689eba/src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java#L325-L328
